homeassistant/components/conversation/default_agent.py

```diff
@@ -60,6 +62,14 @@ class LanguageIntents:
     loaded_components: set[str]
 
 
+@dataclass(slots=True)
+class TriggerData:
+    """List of sentences and the callback for a trigger."""
+
+    sentences: list[str]
+    callback: TRIGGER_CALLBACK_TYPE
```

Comment:
The `callback` should be `async` if we want to add dynamic responses based on something in the automation in the future 🤔 But then we don't really want to wait for a full automation to run either.

Reply:
We can start with it being callbacks today I guess

Reply:
I made it async and am waiting for the automation to complete. No response can be returned yet, but this will make sure the confirmation in ESPHome, etc. is delayed until the automation finishes.